### PR TITLE
Update .golangci.yml (removed channels/api4/license.go from list)

### PR DESCRIPTION
**Pull request by TheInvincibleRalph**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28401

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/.golangci.yml [hunk 1]
```diff
@@ -32,28 +32,28 @@ issues:
     - channels/store/storetest/mocks
   exclude-rules:
     - linters:
-      # ignore unused warnings from enterprise code
```
Comment (hanzei) [inline server/.golangci.yml:35]: Can you please revert these unrelated changes?

File: server/channels/api4/emoji_test.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ func TestCreateEmoji(t *testing.T) {
 	// try to create an emoji when they're disabled
 	_, resp, err := client.CreateEmoji(context.Background(), emoji, utils.CreateTestGif(t, 10, 10), "image.gif")
 	require.Error(t, err)
+	require.NotNil(t, err)
```
Comment (hanzei) [inline server/channels/api4/emoji_test.go:64]: `require.NoError` in the previous line already checks the error.

File: server/channels/api4/emoji_test.go [hunk 3]
```diff
@@ -509,6 +524,7 @@ func TestDeleteEmoji(t *testing.T) {
 	th.AddPermissionToRole(model.PermissionDeleteOthersEmojis.Id, model.TeamUserRoleId)
 
 	client.Logout(context.Background())
```
Comment (hanzei) [inline server/channels/api4/emoji_test.go:526]: You need the assign the return value to `err`.

File: server/channels/api4/emoji_test.go [hunk 4]
```diff
@@ -795,9 +823,11 @@ func TestAutocompleteEmoji(t *testing.T) {
 
 	_, resp, err = client.AutocompleteEmoji(context.Background(), "", "")
 	require.Error(t, err)
+	require.NotNil(t, resp, "Expected a non-nil response")
 	CheckBadRequestStatus(t, resp)
 
-	client.Logout(context.Background())
+	_, err = client.Logout(context.Background())
+	require.NoError(t, err)
```
Comment (hanzei) [inline server/channels/api4/emoji_test.go:830]: This is correct :+1:

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR @TheInvincibleRalph :+1:

**Comment by TheInvincibleRalph** [comment]
Thank you for the review, @hanzei. I'll work on it and get back. Thank you!

**Comment by TheInvincibleRalph** [comment]
Made corrections based on your review, @hanzei. Thank you so much!

**Review by hanzei — CHANGES_REQUESTED** [review]
The changes in `server/channels/api4/emoji_test.go` look good.

Can you please revert the unrelated changes in `server/.golangci.yml`? Thanks!

File: server/.golangci.yml (hunk 1 above)
Comment (TheInvincibleRalph) [reply server/.golangci.yml:35]: Done. Thank you.
